### DISCUSSION TITLE
Cannot mount block storage to pod: Error: map failed: (30) Read-only file system

### DISCUSSION THREAD

**Discussion by batra98** [rook/rook#9252] [Q&A]
- I have been trying to deploy a ceph cluster using rook with the help of the following tutorial: https://rook.io/docs/rook/v1.7/quickstart.html.
- When I create a block storage and attach it to the sample Wordpress application as given here: https://rook.io/docs/rook/v1.7/ceph-block.html, I get the following error:
``` 
Warning FailedMount            1s (x2 over 1s) kubelet                 MountVolume.MountDevice failed for volume "pvc-7e4a98a0-4476-43e1-a0
0c-2d315a62455d" : rpc error: code = Internal desc = rbd: map failed with error an error (exit status 30) occurred while running rbd args: [--id
csi-rbd-node -m 10.108.116.211:6789 --keyfile=***stripped*** map replicapool/csi-vol-6d934fe3-4dc6-11ec-89a3-0242ac110004 --device-type krbd --op
tions noudev], rbd error output: rbd: sysfs write failed
rbd: map failed: (30) Read-only file system
```
- This is somewhat similar to this: https://github.com/kubernetes-sigs/kind/issues/745. 
- Also, cephfs works fine in my case, just the block-storage gives errors.

**Comment by travisn** [comment]
@batra98 Did you use all the default settings in the quickstart guide? Or did any settings change? 
@Rakshith-R ideas?

**Reply by Rakshith-R** [reply]
hey @batra98 , 
Can you take a look at https://github.com/ceph/ceph-csi/issues/2397 , 
its a similar issue.
Can you share more details about the environment? 
kernel version, kubernetes version, minikube version ?

**Reply by batra98** [reply]
Hi @Rakshith-R,
Thanks for replying. I will go through the issue.
 Here are some details of my working environment:
- Minikube
`minikube version: v1.24.0`
`commit: 76b94fb3c4e8ac5062daf70d60cf03ddcc0a741b`

- Kubernetes
```Client Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:41:28Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
Server Version: version.Info{Major:"1", Minor:"22", GitVersion:"v1.22.3", GitCommit:"c92036820499fedefec0f847e2054d824aea6cd1", GitTreeState:"clean", BuildDate:"2021-10-27T18:35:25Z", GoVersion:"go1.16.9", Compiler:"gc", Platform:"linux/amd64"}
```

- Kernel-Version: I am working on an azure server.
```5.4.0-1063-azure```

@travisn 
I followed the guide in rook's documentation, just used `cluster-test.yaml` instead of `cluster.yaml`.
Also, I recently checked that the object store also works perfectly.
